Fix integration config validation

hluchas opened the pull request:
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.1 for bug fixes
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | -
| Related developer documentation PR URL | -
| Issue(s) addressed                     | -

#### Description:
When debugging internal plugin we discovered a bug in integration config form validation. Is possible to call not existing method `isAuthorized` when `ConfigFormAuthorizeButtonInterface` is not used. This is clearly visible in changes, but difficult to reproduce.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. You 'll need to use some plugin implementing `ConfigFormAuthorizeButtonInterface::isAuthorized` method.
3. Remove usage of `ConfigFormAuthorizeButtonInterface` via implements clause and method `isAuthorized` too..
4. Try to save config.
5. No matter what happens then but no exception `Internal Server Error - Attempted to call an undefined method named "isAuthorized" of class ` is not thrown.

hluchas committed:
Fix condition when to validate plugin config form

hluchas committed:
Add test case

hluchas committed:
Mark test ase and skipped with described reason in comment

hluchas committed:
Switch test cases order

hluchas committed:
Fix api keys params

hluchas committed:
Add test

hluchas committed:
Remove test to test failing build

hluchas committed:
Fix condition to use shorter syntax

shinde-rahul commented on app/bundles/IntegrationsBundle/Controller/ConfigController.php:
```diff
@@ -152,7 +152,10 @@ private function submitForm()
 
         // Show the form if there are errors
-        if (!$this->form->isValid()) {
+        if (!$this->form->isValid() && (!$this->integrationObject instanceof ConfigFormAuthorizeButtonInterface || $this->integrationObject->isAuthorized())) {
```
This will fail for the following scenario,
When the plugin is configured the first time and the user tries to save the configuration without entering values in the required fields. It will not throw the NotBlank constraint even if `!$this->form->isValid()` is true. As `isAuthorized` will always send false for fresh configuration.

hluchas replied:
I can't reproduce with empty plugin configuration
<img width="570" alt="Snímek obrazovky 2020-11-20 v 15 12 00" src="https://user-images.githubusercontent.com/12815758/99810891-acb14000-2b44-11eb-97aa-99433d909dc2.png">

Are you able to record video or describe which plugin was used and write exact steps?
TY!

alanhartless committed:
Better sanitization of URLs

hluchas committed:
Fix URL

hluchas committed:
Add test

hluchas committed:
Fix typehint

dennisameling committed:
Add and improve input cleaning in multiple areas

johbuch committed:
Fix dateandtime defaultvalue (#9534)

* change method validateValue() to validate()

* re added annotations

Co-authored-by: Norman Pracht - Webmecanik <npr@webmecanik.com>

escopecz committed:
Run composer test command with 4G (#9523)

* Running PHPUNIT tests needs more memory than the server might be configured for. Set it for the command itself

* 4G will be better

Co-authored-by: Miroslav Fedeleš <miroslav.fedeles@gmail.com>

Co-authored-by: Miroslav Fedeleš <miroslav.fedeles@gmail.com>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

kuzmany committed:
Fix getSecondaryIdentifier for Lead (#9444)

* Fix getSecondaryIdentifier

* Not display getSecondaryIdentifier If company is in columns

Co-authored-by: Norman Pracht - Webmecanik <npr@webmecanik.com>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

RCheesley committed:
Remove outdated issue template

RCheesley committed:
Update email address

RCheesley committed:
Revert "Remove outdated PR template"

This reverts commit a88e83cd

RCheesley committed:
Update README.md

maxlawton committed:
Support graceful timeout option on ConsumeQueueCommand (#9394)

Added an option (--timeout=TIMEOUT, -t TIMEOUT) takes a number of
seconds to be used as the graceful execution time on the Queue
Consumer.

The value is then passed as an argument to the Queue Service's
consumeFromQueue method, where it is passed as a constructor argument
for the QueueEvent to be dispatched. Then, the RabbitMqSubscriber uses
the event's timeout property to maybe call the consumer's
setGracefulMaxExecutionDateTimeFromSecondsInTheFuture method before
calling $consumer->consume(...).

This will limit the execution time of the ConsumeQueueCommand by
allowing it to exit gracefully after TIMEOUT seconds have elapsed.

dennisameling committed:
Update version to 3.2.3

dennisameling committed:
Fix update error detection logic

dennisameling committed:
Bump version to 3.2.4

dennisameling committed:
Fix Codecov badge in README.md

dennisameling committed:
Fix return type on ConfigControllerFunctionalTest

dennisameling reviewed (changes requested):
While this looks good from a code perspective, the tests are failing. It also seems rather complex to create a plugin that implements `ConfigFormAuthorizeButtonInterface`; at least I don't have a single plugin at my disposal that uses this interface.

@hluchas any chance you could share the plugin that's causing trouble with me privately? I assume it's called `Salesforce2` because that's what you used in the automated test as well 😉

dev-marcel committed:
Fix for issue #8365 (#8746)

Co-authored-by: Norman Pracht - Webmecanik <npr@webmecanik.com>

npracht commented:
@hluchas can you have a look at tests issue please ?

hluchas commented:
> @hluchas can you have a look at tests issue please ?

It is different plugin. Taking a look at this

hluchas committed:
Remove functional tests covering misiing behavior

hluchas commented:
I removed the functional test as fast solution for now. I didn't know how to get it work with other plugin.

kuzmany commented:
@hluchas could you rebase it to features branch. I would like test it

hluchas commented:
I rebased staging to this branch and there's still problem with 
```
.github/ISSUE_TEMPLATE.md
app/release_metadata.json
```

hluchas commented:
Okay, I used staging for rebase. I'll fix this with other PR :-/

hluchas commented:
Fixed here https://github.com/mautic/mautic/pull/9642
